format/src/main/java/com/github/sadikovi/riff/FileWriter.java

```diff
@@ -262,6 +265,7 @@ public void write(InternalRow row) throws IOException {
       updateColumnFilters(stripeFilters, td, row);
       recordWriter.writeRow(row, stripeStream);
       stripeCurrentRecords--;
+      totalRecords++;
```

Comment:
It might be better to update count based on total records written in stripe, rather than increment for every record.